modules/api/test/ApiTest.php

```diff
@@ -0,0 +1,392 @@
+<?php declare(strict_types=1);
+/**
+ * This contains tests relevant to the API
+ *
+ * PHP Version 7
+ *
+ * @category   API
+ * @package    Tests
+ * @subpackage Login
+ * @author     Simon Pelletier <simon.pelletier@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+
+
+namespace LORIS\api\Test;
+ini_set('memory_limit', '1024M');
+
+require './vendor/autoload.php';
+require_once __DIR__ .
+    "/../../../test/integrationtests/LorisIntegrationTest.class.inc";
+
+use phpDocumentor\Reflection\Types\AbstractList;
+use phpDocumentor\Reflection\Types\Array_;
+use PHPUnit\Framework\TestCase;
+use Laminas\Diactoros\ServerRequest;
+use GuzzleHttp\Client;
+
+/**
+ * PHPUnit class for API testsuite
+ *
+ * @category   API
+ * @package    Tests
+ * @subpackage Login
+ * @author     Simon Pelletier <simon.pelletier@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+
+
+class ProjectsTest extends TestCase
+{
+
+    private $_client;
+    /**
+     * @var string[]
+     */
+    private $_headers;
+    private $_base_uri;
+
+    /**
+     * @var string[]
+     */
+
+    private function _login()
+    {
+        $this->_base_uri = 'https://test-loris-dev.loris.ca/api/v0.0.3/';
```

Comment:
What if this VM is down or the host name changed.

Reply:
~@kongtiaowang What is the baseurl when the server is run in TravisCI?~
nerver mind. With LorisIntegrationTest.class.inc we just need to use`$this->url`

Reply:
@spell00 
`$this->_base_uri = $this->url . '/api/v0.0.3/' ` should work